Marlin/src/gcode/host/M115.cpp

```diff
@@ -89,12 +89,12 @@ void GcodeSuite::M115() {
      */
     #if ENABLED(STM32_UID_SHORT_FORM)
       uint32_t * const UID = (uint32_t*)UID_BASE;
-      SERIAL_ECHO(hex_long(UID[0]), hex_long(UID[1]), hex_long(UID[2]));
+      SERIAL_ECHO(strdup(hex_long(UID[0])), strdup(hex_long(UID[1])), hex_long(UID[2]));
```

Comment:
strdup() seems to cause a memory leak, 40 bytes for short form and 80 bytes for long form everytime M115 is executed.

Reply:
strdup removed

Reply:
Strings created with `strdup` must be freed with `free`, but for the most part we avoid any `new`, `malloc`, or equivalent memory allocation because it can lead to fragmentation and subsequent failure to allocate buffers, especially on memory-starved boards.